src/client_features.rs

```diff
@@ -285,6 +300,31 @@ pub struct ClientFeature {
     pub variants: Option<Vec<Variant>>,
 }
 
+impl Merge for ClientFeatures {
+    fn merge(self, other: Self) -> Self {
+        let features = self.features.merge(other.features);
+
+        let segments = match (self.segments, other.segments) {
+            (Some(mut s), Some(o)) => {
+                s.extend(o);
```

Comment:
Probably a dumb question, but is this thread-safe (in case it needs to be)? We're mutating an existing segment array that might be used by another thread.

Reply:
It is, we're consuming both here, so we're guaranteed to be the only one that can do this at this point in time.